File: backend/src/main/java/com/campus/delivery/modules/order/entity/Order.java
package com.campus.delivery.modules.order.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.Version;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 订单主表。主责：成员4。
 *
 * <p>{@code orderStatus} 只能通过 {@code OrderStateMachine} 变更，
 * {@code version} 为乐观锁版本号，防止并发下的非法状态写入。
 */
@Data
@TableName("orders")
public class Order implements Serializable {

    @TableId(value = "id", type = IdType.ASSIGN_UUID)
    private String id;

    /** 对外展示的订单编号 */
    private String orderNo;

    private String userId;

    private String shopId;

    private String riderId;

    private BigDecimal goodsAmount;

    private BigDecimal deliveryFee;

    private BigDecimal totalAmount;

    /** 1余额 2模拟第三方 */
    private Integer payType;

    /** 0未支付 1已支付 2已退款 */
    private Integer payStatus;

    private LocalDateTime payTime;

    /** 0待接单 1备餐中 2待取餐 3配送中 4已完成 5已取消 6申诉中 */
    private Integer orderStatus;

    /** 1骑手配送 2宿舍楼下自取 */
    private Integer deliveryType;

    private String receiver;

    private String receiverPhone;

    private String address;

    private String remark;

    private String pickupCode;

    private String cancelReason;

    private LocalDateTime expectTime;

    private LocalDateTime finishTime;

    @Version
    private Integer version;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}

File: frontend/src/api/order.ts
import { get, post } from '@/utils/request'
import type { PageResult } from '@/utils/request'

/** 订单状态：0 待接单 1 备餐中 2 待取餐 3 配送中 4 已完成 5 已取消 6 申诉中 */
export interface Order {
  id: string
  orderNo: string
  userId?: string
  shopId?: string
  riderId?: string
  goodsAmount?: number
  deliveryFee?: number
  totalAmount: number
  orderStatus: number
  payStatus?: number
  payType?: number
  deliveryType?: number
  receiver?: string
  receiverPhone?: string
  address?: string
  remark?: string
  pickupCode?: string
  createTime?: string
}

export interface OrderItem {
  id?: string
  dishId: string
  dishName: string
  dishImage?: string
  price: number
  quantity: number
  amount: number
}

export interface OrderVO {
  order: Order
  items: OrderItem[]
  statusDesc?: string
}

export interface CreateOrderParams {
  deliveryType?: number
  payType?: number
  receiver: string
  receiverPhone: string
  address?: string
  remark?: string
}

/** 学生端 */
export function createOrder(data: CreateOrderParams) {
  return post<OrderVO>('/orders', data)
}

export function listMyOrders(params?: { status?: number; pageNum?: number; pageSize?: number }) {
  return get<PageResult<OrderVO>>('/orders', params)
}

export function getOrderDetail(id: string) {
  return get<OrderVO>(`/orders/${id}`)
}

export function cancelOrder(id: string, reason?: string) {
  return post<void>(`/orders/${id}/cancel`, { reason })
}

/** 商户端 */
export function listShopOrders(params?: { status?: number; pageNum?: number; pageSize?: number }) {
  return get<PageResult<OrderVO>>('/merchant/orders', params)
}

export function acceptOrder(id: string) {
  return post<Order>(`/merchant/orders/${id}/accept`)
}

export function readyOrder(id: string) {
  return post<Order>(`/merchant/orders/${id}/ready`)
}

export function rejectOrder(id: string, reason?: string) {
  return post<Order>(`/merchant/orders/${id}/reject`, { reason })
}
